src/mudlet-lua/lua/Other.lua

```diff
@@ -700,3 +700,111 @@ function condenseMapLoad()
 
   return loadtime
 end
+
+do
+  -- management things
+
+  -- Dictionary with events as keys and lists of lua functions as values to dispatch events
+  -- to the right functions.
+  local handlers = {}
+
+  -- Remember highest hander ID to avoid ID reuse.
+  local highestHandlerId = 0
+  -- Helps us finding the right event handler from an ID.
+  local handlerIdsToHandlers = {}
+
+  -- C functions that get overwritten.
+  local origRegisterAnonymousEventHandler = registerAnonymousEventHandler
+
+  function registerAnonymousEventHandler(event, func, isOneShot)
+    if type(event) ~= "string"then
+      error(
+        string.format(
+          "Unexpected argument type in function registerAnonymousEventHandler for argument #1. String expected, got %s.",
+          type(event)
+        )
+      )
+    end
+
+    if type(func) ~= "function" and type(func) ~= "string" then
+      error(
+        string.format(
+          "Unexpected argument type in function registerAnonymousEventHandler for argument #2. Function or string expected, got %s.",
+          type(func)
+        )
+      )
+    end
+
+    if type(func) == "string" then
+      func = assert(loadstring(string.format("%s(...)", func)))
```

Comment:
Does this compile `string` and if so what happens if it fails - do we need to report something (in our current run-time "bad-value" form which does not now refer to the function or the argument concerned as I once tended to do in the C++ code):

    "\"func\" did not compile correctly."

?

Reply:
`loadstring` compiles the string within, correct. The assert around it will check the first return value for a `true-thy` value and return everything if it is so. If the first returned value is `false-y`, it assumes that the second returned value is an error string and hands that to the `error()` function. See https://www.lua.org/pil/8.3.html